### PR TITLE
docs: how to setup pipeline for Azure DevOps

### PR DESCRIPTION
**Pull request by lesyk**
Add: documentation on how to setup pipeline for Azure DevOps that will be running renovate once a day.

### CODE REVIEW

File: docs/usage/setup-azure-devops.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    displayName: 'Every day at 3am'
+    branches:
+      include:
+        - master
+    always: true
+
+trigger: none
+
+variables:
+  SYSTEM_ACCESSTOKEN: $(System.AccessToken)
```
Comment (JamieMagee) [inline docs/usage/setup-azure-devops.md:32]: This was left over from troubleshooting?

File: docs/usage/setup-azure-devops.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            module.exports = {
+              platform: 'azure',
+              endpoint: 'https://dev.azure.com/PLACEHOLDER(ORGANIZATION)/',
+              token: '$(System.AccessToken)',
+              azureWorkItemId: PLACEHOLDER,
+              dryRun: false,
+              includeForks: true,
+              logLevel: 'debug',
+              onboarding: false,
+              requireConfig: false,
```
Comment (JamieMagee) [inline docs/usage/setup-azure-devops.md:70]: I'm not sure these options are necessary for a basic pipeline.

File: docs/usage/setup-azure-devops.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+              npmrc:
+                '$(sed ':a;N;$!ba;s/\n/\\n/g' .npmrc)',
+              ignoreNpmrcFile: true,
+              hostRules: [
+                {
+                  hostName: 'pkgs.dev.azure.com',
+                  username: 'renovate',
+                  password: '$(System.AccessToken)'
+                }
+              ],
```
Comment (JamieMagee) [inline docs/usage/setup-azure-devops.md:80]: I don't think this is required for JS projects

File: docs/usage/setup-azure-devops.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+![Azure DevOps create new pipeline](assets/images/azure-devops-setup-1.png)
+
+1. Select your repository.
+
+1. Within <i>Configure your pipeline</i> select: <b>Starter pipeline file</b>.
+
+![Azure DevOps starter pipeline template](assets/images/azure-devops-setup-2.png)
+
+1. Replace all contents with next template (change all <i>PLACEHOLDER</i> strings with your values):
```
Comment (JamieMagee) [inline docs/usage/setup-azure-devops.md:18]: Is it worth mentioning that this template is specific to npm/yarn projects?

**Comment by JamieMagee** [comment]
@rarkins @viceice This is adapted from a configuration I've shared with some colleagues. @lesyk cleaned it up, and made this documentation page out of it.

**Review by viceice — CHANGES_REQUESTED** [review]
I think we should prefer a docker image based approach, so any dependency type supported by renovate can be used.

Also the **CLA** needs to be signed

File: docs/usage/setup-azure-devops.md (hunk 4 above)
Comment (massto) [reply docs/usage/setup-azure-devops.md]: > Is it worth mentioning that this template is specific to npm/yarn projects?

**Comment by viceice** [comment]
@lesyk Can you please sign the CLA. Siply use the link from the cla image above or go [here](https://cla-assistant.io/renovatebot/renovate?pullRequest=6929)

**Comment by rarkins** [comment]
@JamieMagee A docs-only contribution is ok without a CLA. Is the PR otherwise good to merge?

**Comment by JamieMagee** [comment]
@rarkins Let me ping @lesyk offline to ask about the CLA.

Apart from that, @viceice is right, and a Docker based approach would be better overall. Though `npx` should be enough for npm/yarn projects.

**Pull request merged**